barretenberg/cpp/src/barretenberg/vm2/constraining/relations/ecc.test.cpp

```diff
@@ -0,0 +1,393 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+
+#include "barretenberg/vm2/constraining/testing/check_relation.hpp"
+#include "barretenberg/vm2/generated/flavor_settings.hpp"
+#include "barretenberg/vm2/generated/relations/ecc.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::constraining {
+namespace {
+
+using tracegen::TestTraceContainer;
+using FF = AvmFlavorSettings::FF;
+using C = Column;
+using ecc = bb::avm2::ecc<FF>;
+using GrumpkinPoint = grumpkin::g1::affine_element;
+
+// Known good points for P and Q
+bb::grumpkin::fq p_x = grumpkin::fq("0x04c95d1b26d63d46918a156cae92db1bcbc4072a27ec81dc82ea959abdbcf16a");
+bb::grumpkin::fq p_y = grumpkin::fq("0x035b6dd9e63c1370462c74775765d07fc21fd1093cc988149d3aa763bb3dbb60");
+GrumpkinPoint p(p_x, p_y);
+
+bb::grumpkin::fq q_x = grumpkin::fq("0x009242167ec31949c00cbe441cd36757607406e87844fa2c8c4364a4403e66d7");
+bb::grumpkin::fq q_y = grumpkin::fq("0x0fe3016d64cfa8045609f375284b6b739b5fa282e4cbb75cc7f1687ecc7420e3");
+GrumpkinPoint q(q_x, q_y);
+
+TEST(AvmConstrainingTest, EccPositiveEmptyRow)
+{
+    TestTraceContainer trace({
+        { { C::precomputed_clk, 1 } },
+    });
+
+    check_relation<ecc>(trace.as_rows());
+}
+
+TEST(AvmConstrainingTest, EccPositiveAdd)
+{
+    // R = P + Q;
+    bb::grumpkin::fq r_x = grumpkin::fq("0x2b01df0ef6d941a826bea23bece8243cbcdc159d5e97fbaa2171f028e05ba9b6");
+    bb::grumpkin::fq r_y = grumpkin::fq("0x0cc4c71e882bc62b7b3d1964a8540cb5211339dfcddd2e095fd444bf1aed4f09");
+    GrumpkinPoint r(r_x, r_y);
+
+    TestTraceContainer::RowTraceContainer trace = { {
```

Comment:
Please use `TestTraceContainer` itself in tests, otherwise the trace does not have shifts set. It might not matter if your relation doesnt use shifts but it's best practice (Jean ran into problems).

A quick change is to do

```
auto trace = TestTraceContainer::from_rows({
// exactly what you have
});

check_relatiion<...>(trace.as_rows());
```

Please change everywhere else.